lua/roslyn/init.lua

```diff
@@ -159,37 +142,51 @@ local function get_default_capabilities(roslyn_config)
     })
 end
 
-local M = {}
+---@param exe string|string[]
+local function get_cmd(exe)
+    local default_lsp_args =
+        { "--logLevel=Information", "--extensionLogDirectory=" .. vim.fs.dirname(vim.lsp.get_log_path()) }
+    local mason_installation = get_mason_installation()
+
+    if type(exe) == "string" then
```

Comment:
I don't think this will work properly. I also looked into using `unpack` when doing the mason support, but it didn't work for me. Also this doesn't support the default case.

I was thinking about something like this:

```lua
---@param exe? string[]
local function get_cmd(exe)
    local default_lsp_args =
        { "--logLevel=Information", "--extensionLogDirectory=" .. vim.fs.dirname(vim.lsp.get_log_path()) }
    local mason_installation = get_mason_installation()

    if type(exe) == "table" then
        return vim.list_extend(vim.deepcopy(exe), default_lsp_args)
    elseif vim.uv.fs_stat(mason_installation) then
        return vim.list_extend({ mason_installation }, default_lsp_args)
    else
        return vim.list_extend({
            "dotnet",
            vim.fs.joinpath(
                vim.fn.stdpath("data") --[[@as string]],
                "roslyn",
                "Microsoft.CodeAnalysis.LanguageServer.dll"
            ),
        }, default_lsp_args)
    end
end
```

I was actually thinking of just dropping support for `string` as argument to cmd so I didn't have it in this small example. However, it seemed to be quite simple and not much extra code, so feel free to add that if you want😄